client/components/data/page/index.jsx

```diff
@@ -0,0 +1,97 @@
+/**
+ * External dependencies.
+ */
+import { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import each from 'lodash/each';
+
+/**
+ * Internal dependencies.
+ */
+import {
+	setTitle,
+	setDescription,
+	addLink,
+	addMeta,
+	setUnreadCount
+} from 'state/page/actions';
+
+class Page extends Component {
+	componentWillMount() {
+		this.props.setTitle( this.props.title );
+		this.props.setDescription( this.props.description );
+		this.props.setUnreadCount( this.props.unreadCount );
+
+		if ( this.props.link && this.props.link.length ) {
+			each( this.props.link, ( link ) => {
+				this.props.addLink( link );
+			} );
+		}
+
+		if ( this.props.meta && this.props.meta.length ) {
+			each( this.props.meta, ( meta ) => {
+				this.props.addMeta( meta );
+			} );
+		}
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( this.props.title !== nextProps.title ) {
+			this.props.setTitle( nextProps.title );
+		}
+
+		if ( this.props.description !== nextProps.description ) {
+			this.props.setDescription( nextProps.description );
+		}
+
+		if ( this.props.unreadCount !== nextProps.unreadCount ) {
+			this.props.setUnreadCount( nextProps.unreadCount );
+		}
+
+		if ( this.props.link !== nextProps.link && this.props.link.length ) {
```

Comment:
I'll not let it hold up the merge, especially as this component isn't yet used anywhere, but strict comparison for arrays is ripe for trouble, both because arrays can be mutated while still keeping their same reference, and because two instances of an array with the same members are strictly unequal.

![image](https://cloud.githubusercontent.com/assets/1779930/14047111/23bdddcc-f27c-11e5-84d3-f8b6d19f262e.png)

![image](https://cloud.githubusercontent.com/assets/1779930/14047120/2d78a054-f27c-11e5-94a6-598d7a92e734.png)

Instead, you'll want to consider using a utility for either shallow or deep comparison. In our project, this means:

- [`react-pure-render/shallowEqual`](https://github.com/gaearon/react-pure-render#shallowequal) for shallow equality
- [`_.isEqual`](https://lodash.com/docs#isEqual) for deep equality

(The latter is probably what you want for comparing an array of objects, as objects suffer similar strict equality issues as arrays)